src/main/io/osd.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define MAX_MENU_ROWS 8
+#define MAX_MENU_COLS 3
+
+typedef struct {
+    const char* title;
+    uint8_t x_pos;
+} col_t;
+
+typedef struct {
+    const char* title;
+    void (*update)(bool increase, uint8_t col);
+    void (*print)(uint16_t pos, uint8_t col);
+} row_t;
+
+typedef struct {
+    const char* title;
+    uint8_t     cols_number;
+    uint8_t     rows_number;
+    col_t       cols[MAX_MENU_COLS];
+    row_t       rows[MAX_MENU_ROWS];
+} page_t;
+
+
+typedef enum {
+    OSD_MAIN_BATT_VOLTAGE,
+    OSD_RSSI_VALUE,
+    OSD_TIMER,
+    OSD_THROTTLE_POS,
+    OSD_CPU_LOAD,
+    OSD_VTX_CHANNEL,
+    OSD_VOLTAGE_WARNING,
+    OSD_ARMED,
+    OSD_DISARMED,
+    OSD_MAX_ITEMS, // MUST BE LAST
+} osd_items;
```

Comment:
`osdItems_t`

also, I'm not sure if we want to put the full list of osd positions in the config, so we dont need to bump the config version and delete the whole config each time one is added?

```
enum MAX_OSD_FIELDS {
  MAX_OSD_FIELD_GPS_NUM_SAT,
  MAX_OSD_FIELD_GPS_TIME,
  MAX_OSD_FIELD_GPS_DIRECTION_HOME,
  MAX_OSD_FIELD_GPS_DISTANCE_HOME,
  MAX_OSD_FIELD_GPS_SPEED,
  MAX_OSD_FIELD_GPS_ANGLE_HOME,
  MAX_OSD_FIELD_GPS_ALTITUDE,
  MAX_OSD_FIELD_GPS_HEADING,
  MAX_OSD_FIELD_GPS_HEADING_GRAPH,
  MAX_OSD_FIELD_GPS_LAT,
  MAX_OSD_FIELD_GPS_LON,
  MAX_OSD_FIELD_MAP_MODE,
  MAX_OSD_FIELD_MAP_CENTER,
  MAX_OSD_FIELD_BARO,
  MAX_OSD_FIELD_CLIMB,
  MAX_OSD_FIELD_PITCH_ANGLE,
  MAX_OSD_FIELD_ROLL_ANGLE,
  MAX_OSD_FIELD_SENSOR,
  MAX_OSD_FIELD_THROTTLE,
  MAX_OSD_FIELD_MOTOR_ARMED,
  MAX_OSD_FIELD_RSSI,
  MAX_OSD_FIELD_TEMP,
  MAX_OSD_FIELD_VOLTAGE,
  MAX_OSD_FIELD_AMPERAGE,
  MAX_OSD_FIELD_MODE,
  MAX_OSD_FIELD_POWER_METER,
  MAX_OSD_FIELD_WATT,
  MAX_OSD_FIELD_TIME,
  MAX_OSD_FIELD_HORIZON,
  MAX_OSD_FIELD_HORIZON_SIDEBARS,
  MAX_OSD_FIELD_GIMBAL,
  MAX_OSD_FIELD_SPORT,
  MAX_OSD_FIELD_CALLSIGN,
  MAX_OSD_FIELD_DEBUG,
  // Count of fields
  MAX_OSD_DISPLAY_FIELDS
};
```